### PR TITLE
fix(api): add hour to metata schema

### PR DESCRIPTION
**Pull request by Orta21**
Ref. metriport/metriport-internal#790

Ref: 790

### Dependencies

- Upstream: https://github.com/metriport/metriport-ios-sdk/pull/7

### Description

Add hour to metadata schema for hourly webhook payloads

### Release Plan

Update api package to version 4.1.2

- [ ] Upstream dependencies are met
- [ ] Once approved

### CODE REVIEW

File: api/app/src/mappings/apple/index.ts [hunk 1]
```diff
@@ -16,35 +19,36 @@ export type AppleWebhookPayload = {
...
 
   return payload;
 }
 
-export const createMetadata = (date: string) => {
+export const createMetadata = (date: string, hourly: boolean): Metadata => {
   return {
-    date,
+    date: dayjs(date).format(ISO_DATE),
+    ...(hourly ? { hour: dayjs(date).format("HH:mm:ss") } : undefined),
```
Comment (Goncharo) [inline api/app/src/mappings/apple/index.ts:51]: If this signifies the hour, should we just have this to always be `HH:00`?

mm:ss might be confusing, especially if they're diff between payloads

File: docs/devices-api/more-info/models.mdx [hunk 2]
```diff
@@ -340,7 +340,9 @@ export interface LatLon {
 ```typescript
 export interface Metadata {
   date: string;
+  hour?: string;
```
Comment (Goncharo) [inline docs/devices-api/more-info/models.mdx:343]: We'll need to update all of our docs to include this as an optional field with a description of expected format - right?

**Review by Goncharo — CHANGES_REQUESTED** [review]

File: docs/devices-api/more-info/apple-health.mdx [hunk 3]
```diff
@@ -138,6 +138,12 @@ This will be a daily average of all data types.
 **Step 2:** Once this data is queried we will then listen on an hourly basis for any updates to the Apple Health.
 If there are changes then a query will be sent via webhook with just the data that's been added.
 
+<Info>
+  When listening for hourly changes you'll notice that the metadata contains an `hour` field. This
+  is to signify the hour of the day in which the data is aggregated as shown in the [
+  model](/devices-api/more-info/models#metadata).
+</Info>
```
Comment (Goncharo) [inline docs/devices-api/more-info/apple-health.mdx:145]: Let's update this to show an example metadata payload & be explicit about what format the `hour` will be in.

**Review by Goncharo — CHANGES_REQUESTED** [review]

**Review by Goncharo — APPROVED** [review]
:shipit:

**Comment by Orta21** [comment]
Created a new PR and cherry picked commits

**Pull request closed without merging**